### ISSUE TITLE
Function IsFileExist() in sema-linux/lib/boardinfo.c leaks file descriptors

### ISSUE THREAD

**Issue by kriskazmar** [ADLINK/sema-linux#5]
Function IsFileExist() in sema-linux/lib/boardinfo.c leaks file descriptors.  This occurs when reading the current CPU temperature for the nanoX-EL using the SEMA EC library with tag "rel_sema4_ec_1v3".  "close(fd);" should be prior to "return EAPI_STATUS_SUCCESS;" and the if should be if (fd < 0).

**Comment by kriskazmar** [comment]
```
uint32_t IsFileExist(char *sysf)
{
	int fd;
	fd = open(sysf, O_RDONLY);
	if (fd < 0)
		return EAPI_STATUS_READ_ERROR;
        close(fd);
	return EAPI_STATUS_SUCCESS;
}
```

**Comment by AdlinkCCoE** [comment]
Added fixes for file descriptor leaks in release 4.3.3
